### ISSUE TITLE
Not clear that some configurations such as [raftstore] raftdb-path are ignored when raft engine is on (default)

### ISSUE THREAD

**Issue by anish-db** [pingcap/docs#13231]
File: [/master/tikv-configuration-file.md](https://docs.pingcap.com/tidb/dev/tikv-configuration-file)

We spent sometime configuring `raftdb-path` in `[raftstore]` before realizing that this parameter was ignored due to raft engine being on by default and a different config had to be used.

The `[raft-engine] enabled` has a short note saying it's on by default and when it's on raftdb configs will be ignored, but it's not clear which all configs are raftdb specific. It would be clearer to add a note next to the config parameters or sections which are ignored that those won't apply unless raft-engine is off (especially since those sections will mostly be ignored with the raft engine default)

A little more confusing was that setting raftdb-path was actually getting validated in info logs (and would fail if we specified a bad dir), making it look it was used, but eventually not actually being used.

**Comment by tabokie** [comment]
I agree the doc can be improved. Specifically "raftdb-path" and the whole "raftdb" section need additional comments on this behavior.

The validation logic adds to the confusion. But it is still necessary in certain cases, e.g. the path will be used when transferring logs from raftdb to raft-engine.
